src/vm/i386/cgenx86.cpp

```diff
@@ -402,6 +402,11 @@ void HelperMethodFrame::UpdateRegDisplay(const PREGDISPLAY pRD)
 
     ClearRegDisplayArgumentAndScratchRegisters(pRD);
 
+    //
+    // Fix up ResumeSp
+    //
+    pRD->pCurrentContext->ResumeEsp = EECodeManager::GetResumeSp(pRD);
```

Comment:
I think it will be more obvious after you make the change that you've suggested.